examples/replay-buffers/compressed_gpu_decompressed_gpu_replay_buffer.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+"""
+Example demonstrating the use of CompressedStorage for memory-efficient replay buffers on the GPU.
+"""
+
+from typing import Any, Dict, List, NamedTuple
+
+import ale_py
+import gymnasium as gym
+
+import numpy as np
+
+gym.register_envs(ale_py)
+import torch
+from torchrl.data import ListStorage, ReplayBuffer
+
+
+class AtariTransition(NamedTuple):
+    """Transition tuple generated by the Atari gymnasium environment."""
+
+    observations: np.uint8
+    actions: np.uint8
+    next_observations: np.uint8
+    rewards: np.float32
+    terminated: np.bool
+    truncated: np.bool
+    info: Dict[str, Any]
+
+
+def main():
+    # pip install gymnasium ale-py opencv-python-headless
+    import time
+
+    import nvidia.nvcomp as nvcomp
+
+    algos = ["Zstd"]  # "LZ4",
+    bitstreams = [
+        nvcomp.BitstreamKind.RAW,
+    ]
+
+    # Create Pong environment and get a frame
+    seed = 42
+    env = gym.make("ALE/Pong-v5", frameskip=1)
+    env = gym.wrappers.AtariPreprocessing(env, frame_skip=5)
+    env = gym.wrappers.RecordEpisodeStatistics(env)
+    env = gym.wrappers.TransformReward(env, np.sign)
+    env = gym.wrappers.FrameStackObservation(env, 4)
+    env.action_space.seed(seed)
+    env.observation_space.seed(seed)
+
+    for algorithm in algos:
+        for bitstream_kind in bitstreams:
+            codec = nvcomp.Codec(algorithm=algorithm, bitstream_kind=bitstream_kind)
+
+            obs, _ = env.reset(seed=0)
+            nv_obs = nvcomp.as_array(obs).cuda(synchronize=False)
+            compressed_obs = codec.encode(nv_obs)
+            decompressed_obs = codec.decode(compressed_obs, data_type="|u1")
+            pt_obs = torch.from_dlpack(decompressed_obs).clone().view(4, 84, 84)
+            assert np.allclose(obs, pt_obs.cpu().numpy())
+
+    print("passed correctness checks")
+
+    # === CompressedListStorage + ReplayBuffer with GPU compression ===
+    print("\n=== ListStorage + ReplayBuffer (GPU) Example ===\n")
+
+    codec = nvcomp.Codec(algorithm=algorithm, bitstream_kind=bitstream_kind)
```

Comment:
I think this deserves a bit of explanation :) 
Also I would check if the lib is installed beforehand and raise a warning /exception if it cannot be found
```
has_nvcomp = importlib.util.findspec("nvcomp", None) is not None
if not has_nvcomp:
    raise ImportError(...)
```